examples/gnrc_tcp_cli/main.c

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (C) 2015 Simon Brummer
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include "net/af.h"
+#include "random.h"
+#include "net/gnrc/ipv6.h"
+#include "net/gnrc/tcp.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#ifndef TARGET_ADDR
+#define TARGET_ADDR "fe80::ccea:ebff:fe12:f132"
+#endif
+
+#ifndef TARGET_PORT
+#define TARGET_PORT 80 /* Target Port */
+#endif
+
+#ifndef CONNS
+#define CONNS 1        /* Number of parallel connections */
+#endif
+
+#ifndef NBYTE          /* Number of parallel connections */
+#define NBYTE (2048)   /* Amount of data to transmit */
+#endif
+
+#ifndef CYCLES
+#define CYCLES 10      /* Number of repeated Cycles per thread */
+#endif
+
+uint8_t bufs[CONNS][NBYTE];
+uint8_t stacks[CONNS][THREAD_STACKSIZE_DEFAULT + THREAD_EXTRA_STACKSIZE_PRINTF];
+
+void *cli_thread(__attribute__((unused)) void *arg);
+
+int main(void)
+{
+    kernel_pid_t ifs[GNRC_NETIF_NUMOF];
+
+    /* Disable 15.4 Retransmissions on the first network device */
+    netopt_enable_t opt = NETOPT_DISABLE;
+    gnrc_netif_get(ifs);
+    gnrc_netapi_set(ifs[0], NETOPT_AUTOACK, 0, &opt, sizeof(opt));
+    gnrc_netapi_set(ifs[0], NETOPT_ACK_REQ, 0, &opt, sizeof(opt));
+
```

Comment:
If you want to use wireshark to figure out whats going on, this is manditory. Will be removed after testing ;).